kubernetes/util.go

```diff
@@ -113,8 +113,8 @@ func IsInCluster(kubeconfig string) bool {
 // If host is provided in the config use it directly.
 // If it is empty then try
 // 1. If beat is deployed in k8s cluster, use hostname of pod as the pod name to query pod metadata for node name.
-// 2. If step 1 fails or beat is deployed outside k8s cluster, use machine-id to match against k8s nodes for node name.
-// 3. If node cannot be discovered with step 1,2, fallback to NODE_NAME env var as default value. In case it is not set return error.
+// 2. If step 1 fails or beat is deployed outside k8s cluster, use NODE_NAME env var.
+// 3. If node cannot be discovered with step 1,2, use machine-id to match against k8s nodes for node name. In case it is not set return error.
```

Comment:
Can we add a note here and a link to the reported issue to raise awareness that this value might not be correct all the times? We have the tribal knowledge here but in the future it would helpful for everyone.